tools/metadata/src/helpers.ts

```diff
@@ -55,5 +53,5 @@ export async function getTokenDecimals(
     throw Error(`Invalid 'tokenAddress' parameter`)
   }
 
-  return getContract(tokenAddress, ERC20_ABI, provider).decimals()
+  return await getContract(tokenAddress, ERC20_ABI, provider).decimals()
```

Comment:
I explained incorrectly my bad, I meant that you can make this function not async, just `export function getTokenDecimals(...)`, and do the normal return as before `return getContract(tokenAddress, ERC20_ABI, provider).decimals()`